node/src/types.ts

```diff
@@ -4,20 +4,28 @@
  * An in-memory representation of a document
  */
 export interface Document {
+  /**
+   * The document identifier
+   */
+  id: string
   /**
    * The absolute path of the document's file.
    */
   path: string
+  /**
+   * Whether or not the document's file is in the temporary directory.
+   */
+  temporary: boolean
+  /**
+   * The synchronization status of the document. This is orthogonal to `temporary` because a document's `content` can be synced or un-synced with the file system regardless of whether or not its `path` is temporary..
+   */
+  status: 'synced' | 'unwritten' | 'unread' | 'deleted'
```

Comment:
@alex-ketch After a little more thought I realized that both `temporary` and `status` are indeed necessary. However, I have removed `status: new` as that was redundant.